astropy/convolution/convolve.py

```diff
@@ -219,8 +219,6 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
 
     if not kernel_sums_to_zero:
         kernel_internal /= kernel_sum
-    else:
-        kernel_internal = kernel
```

Comment:
Anything that occurred before this, e.g. converting to float type ([here](https://github.com/astropy/astropy/pull/7541/files#diff-f632180d9f1e3fcf689f62e4c424e94fR175)) and filling masked kernels ([here](https://github.com/astropy/astropy/pull/7541/files#diff-f632180d9f1e3fcf689f62e4c424e94fR198)) gets undone.